File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.util.Scanner;


public class Main {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        String cal = sc.nextLine();

        String[] str = cal.split("-");

        int result = 0;
        for (int i = 0; i < str.length; i++) {
            int temp = mySum(str[i]);

            if (i == 0){
                result += temp;
            }else{
                result -= temp;
            }
        }
        System.out.println(result);

    }
    public static int mySum(String formula){
        String[] nums = formula.split("[+]");
        int sum = 0;
        for(int i = 0; i<nums.length; i++){
            sum += Integer.parseInt(nums[i]);
        }
        return sum;
    }



}

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
from collections import deque
from typing import List

def returnSum(function : List[str]):
    sumV = 0
    for x in function:
        nums = list(map(int, x.split('+')))
        sumV += sum(nums)

    return sumV;

waits = {'+' : [],
         '-' : []}

operate = input()

values = operate.split('-')

for op in values:
    key_index = operate.find(op)
    if(key_index == 0):
        waits['+'].append(op)
    else:
        if (operate[key_index - 1] == '+'):
            waits['+'].append(op)
        elif (operate[key_index - 1] == '-'):
            waits['-'].append(op)

    operate = operate.replace(op, 'X', 1)

result = 0
for op, function in waits.items():
    temp = returnSum(function)
    if op == '+':
        result += temp
    elif op == '-':
        result -= temp
 
print(result)